opslevel/resource_opslevel_service_repository.go

```diff
@@ -282,5 +282,74 @@ func (r *ServiceRepositoryResource) Delete(ctx context.Context, req resource.Del
 }
 
 func (r *ServiceRepositoryResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
-	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+	ids := strings.SplitN(req.ID, ":", 2)
+	if len(ids) != 2 {
+		resp.Diagnostics.AddError(
+			"Invalid format for given Import Id",
+			fmt.Sprintf("Id expected to be formatted as '<service-id-or-alias>:<repository-id-or-alias>'. Given '%s'", req.ID),
+		)
+		return
+	}
+
+	serviceIdentifier := ids[0]
+	repoIdentifier := ids[1]
+
+	var service *opslevel.Service
+	var err error
+	if opslevel.IsID(serviceIdentifier) {
+		service, err = r.client.GetService(opslevel.ID(serviceIdentifier))
+	} else {
+		service, err = r.client.GetServiceWithAlias(serviceIdentifier)
+	}
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to read service, got error: %s", err))
+		return
+	}
+	repositories, err := service.GetRepositories(r.client, nil)
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to get service dependencies, got error: %s", err))
+		return
+	}
+
+	var foundRepoIdentifier string
+	for _, serviceRepoEdge := range repositories.Edges {
+		if serviceRepoEdge.Node.Id != opslevel.ID(repoIdentifier) && serviceRepoEdge.Node.DefaultAlias != repoIdentifier {
+			continue
+		}
+		for _, serviceRepo := range serviceRepoEdge.ServiceRepositories {
+			if serviceRepo.Repository.Id == opslevel.ID(repoIdentifier) || serviceRepo.Repository.DefaultAlias == repoIdentifier {
+				foundRepoIdentifier = string(serviceRepo.Id)
+				break
+			}
+		}
+		if foundRepoIdentifier != "" {
+			break
+		}
+	}
```

Comment:
So if i'm understanding this logic - this import will take over the "first" service_repository that matches instead of throwing an error when this ambiguous edgecase happens.  I don't hate it because it means it "works more often" but it also sort of creates "unexpected results" because if you really thought you were targeting the 2nd service repository in the list but actually got the first (or our API reorders them in the response) things could get weird.  Thoughts?

Reply:
IMO this strikes a balance between working with the API we have and not asking customers for too many specifics during the import operation. Eventually we could write tests to verify these cases

Reply:
👍